examples/chatgpt/gpt_actions_library/gpt_middleware_azure_function.ipynb

```diff
@@ -0,0 +1,494 @@
+   "metadata": {},
+   "source": [
+    "Check out these links from the application before you get started:\n",
+    "- Application Website: https://learn.microsoft.com/en-us/azure/azure-functions/\n",
+    "- Application API Documentation: https://learn.microsoft.com/en-us/azure/azure-functions/functions-reference/"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Application Prerequisites"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Before you get started, make sure you go through the following steps in your application environment:\n",
+    "- Azure Portal with access to create Azure Function Apps and Azure Entra App Registrations"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Application Setup"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Installing the app"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can read more about languages and deployment options for Azure Functions on the left hand side of the documentation [here](https://learn.microsoft.com/en-us/azure/azure-functions/functions-overview?pivots=programming-language-csharp). "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Option 1: Use VSCode\n",
+    "\n",
+    "See Microsoft’s documentation [here](https://learn.microsoft.com/en-us/azure/azure-functions/functions-develop-vs-code?tabs=node-v4,python-v2,isolated-process\\&pivots=programming-language-javascript) for how to deploy using VSCode. If you have familiarity with this approach, feel free to use it. \n",
+    "\n",
+    "\n",
+    "#### Option 2: Directly in Azure Portal\n",
+    "\n",
+    "See the documentation [here](https://learn.microsoft.com/en-us/azure/azure-functions/functions-create-function-app-portal?pivots=programming-language-javascript) for how to deploy using the Azure portal. We’ll walk through an example here step by step.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Part 1: Create Function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![](../../../images/create_function_app.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. Create an [Azure Function app](https://learn.microsoft.com/en-us/azure/azure-functions/functions-overview?pivots=programming-language-csharp). I used the following settings but you can use anything you are comfortable with. Note that not every language / operating system allows for editing the functions in the console directly - the combination I chose below does. For my walkthrough, I left everything as default and made the selections below\n",
+    "\n",
+    "   1. Basics\n",
+    "\n",
+    "      1. _Do you want to deploy code or container image?:_  **Code**\n",
+    "\n",
```

Comment:
probably worth calling out here that you can change these to others, but these settings will be compatible with the sharepoint cookbooks